### PR TITLE
Created summary cell

### PR DESCRIPTION
**Pull request by lorenzoPrimi**

### CODE REVIEW

**Comment by bassosimone** [comment]
👀

File: ooniprobe/View/TestResults/Rows/TestSummaryTableViewCell.m [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+- (void)awakeFromNib {
+    [super awakeFromNib];
+    // Initialization code
+}
+
+- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
+    [super setSelected:selected animated:animated];
+
+    // Configure the view for the selected state
+}
```
Comment (bassosimone) [inline ooniprobe/View/TestResults/Rows/TestSummaryTableViewCell.m]: The two methods above seem to call a superclass; how about we just omit them?

File: ooniprobe/View/TestResults/TestSummaryTableViewController.m [hunk 2]
```diff
@@ -1,12 +1,14 @@
-#import "TestSummaryTableViewController.h"
+#import "TestSummaryViewController.h"
```
Comment (bassosimone) [inline ooniprobe/View/TestResults/TestSummaryTableViewController.m]: Are you sure this name change is correct?

File: ooniprobe/View/TestResults/TestSummaryTableViewController.m [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 #import "ReachabilityManager.h"
 #import "Tests.h"
+#import "UploadFooterViewController.h"
```
Comment (bassosimone) [inline ooniprobe/View/TestResults/TestSummaryTableViewController.m]: I believe this file is added in another PR, no?

File: ooniprobe/View/TestResults/TestSummaryTableViewController.m [hunk 4]
```diff
@@ -16,16 +18,17 @@ - (void)viewDidLoad {
     [self.navigationController.navigationBar setBackgroundImage:[UIImage new] forBarMetrics:UIBarMetricsDefault];
     self.tableView.tableFooterView = [UIView new];
     [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(resultUpdated:) name:@"resultUpdated" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(reloadMeasurements) name:@"uploadFinished" object:nil];
```
Comment (bassosimone) [inline ooniprobe/View/TestResults/TestSummaryTableViewController.m]: I guess emitting this notification is fine, even though we've not implemented the upload of measurements yet.

Comment (bassosimone) [reply ooniprobe/View/TestResults/TestSummaryTableViewController.m]: Yet, I'd remove this code and reintroduce it later.

File: ooniprobe/View/TestResults/TestSummaryTableViewController.m [hunk 5]
```diff
@@ -44,6 +47,14 @@ - (void)resultUpdated:(NSNotification *)notification
...
     self.measurements = result.measurements;
     dispatch_async(dispatch_get_main_queue(), ^{
+        if ([result isEveryMeasurementUploaded]){
+            self.tableFooterConstraint.constant = -45;
+            [self.tableView setNeedsUpdateConstraints];
+        }
+        else {
+            self.tableFooterConstraint.constant = 0;
+            [self.tableView setNeedsUpdateConstraints];
+        }
```
Comment (bassosimone) [inline ooniprobe/View/TestResults/TestSummaryTableViewController.m]: Also this block of code does not belong here. Please remove.

File: ooniprobe/View/TestResults/TestSummaryTableViewController.m [hunk 6]
```diff
@@ -233,6 +132,11 @@ - (void) prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
         [vc setResult:result];
         [vc setMeasurement:segueObj];
     }
+    else if ([[segue identifier] isEqualToString:@"footer_upload"]){
+        UploadFooterViewController *vc = (UploadFooterViewController * )segue.destinationViewController;
+        [vc setResult:result];
+        [vc setUpload_all:true];
+    }
```
Comment (bassosimone) [inline ooniprobe/View/TestResults/TestSummaryTableViewController.m]: Also this block of code does not belong here; please remove.

**Comment by lorenzoPrimi** [comment]
This can't be merged as it will break master. Storyboard conflict. Also is strictly connect with table rows where I added the upload icon.
Should be reviewed inside https://github.com/ooni/probe-ios/pull/258

**Comment by bassosimone** [comment]
Please, reopen and fix the build. We can cherry pick the commit somewhere else. My job is hopeless if I cannot trust the build system at Travis to do its job.

**Comment by bassosimone** [comment]
I want to do some more on my own on this PR, even if we eventually don't merge it. I think at this point it's important that I feel the pain of the storyboard myself.

**Comment by bassosimone** [comment]
Pain was felt. We can now close this PR.

**Pull request closed without merging**